File: DartLibraryGenerator/sheme_model/api_method.dart
import '../utilities/json_utilities.dart';
import 'api_documentation_node.dart';

class ApiMethod {
  ApiMethod({
    this.parametersId,
    this.accessibleFrom,
    required this.name,
    required this.docs,
    required this.responseId,
    required this.requireAccessToken,
  });

  String name;
  List<ApiDocumentationNode> docs;
  int? parametersId;
  int responseId;
  bool requireAccessToken;
  MethodAccessPolicy? accessibleFrom;

  factory ApiMethod.fromJson(Map<String, dynamic> json) => ApiMethod(
    name: json["name"],
    docs: List<ApiDocumentationNode>.from(json["docs"].map((x) => ApiDocumentationNode.fromJson(x))),
    parametersId: json["parametersId"],
    responseId: json["responseId"],
    requireAccessToken: json["requireAccessToken"],
    accessibleFrom: JsonUtilities.jsonToEnumN(json["accessibleFrom"], MethodAccessPolicy.values),
  );

  Map<String, dynamic> toJson() => {
    "name": name,
    "docs": List<dynamic>.from(docs.map((x) => x.toJson())),
    "parametersId": parametersId,
    "responseId": responseId,
    "requireAccessToken": requireAccessToken,
    "accessibleFrom": JsonUtilities.enumToJson(accessibleFrom),
  };
}

enum MethodAccessPolicy {
  Controller,
  Device,
  Both
}

File: NetLibraryGenerator/NetLibraryGenerator/SchemeModel/ApiMethod.cs
using Ardalis.SmartEnum;
using Ardalis.SmartEnum.JsonNet;
using Newtonsoft.Json;

namespace NetLibraryGenerator.SchemeModel;

public class ApiMethod
{
    [JsonProperty("name")]
    public string Name { get; set; } = null!;
    
    [JsonProperty("docs")]
    public JsDocumentationNode[] Docs { get; set; } = null!;
    
    [JsonProperty("parametersId")]
    public int? ParametersId { get; set; }
    
    [JsonProperty("responseId")]
    public int ResponseId { get; set; }
    
    [JsonProperty("requireAccessToken")]
    public bool RequireAccessToken { get; set; }
    
    [JsonProperty("accessibleFrom")]
    [JsonConverter(typeof(SmartEnumNameConverter<MethodAccessPolicy, int>))]
    public MethodAccessPolicy AccessibleFrom { get; set; } = null!;
}

public class MethodAccessPolicy : SmartEnum<MethodAccessPolicy>
{
    public MethodAccessPolicy(string name, int value): base(name, value) { }
    
    public static readonly MethodAccessPolicy Controller = new MethodAccessPolicy("Controller", 1);
    public static readonly MethodAccessPolicy Device     = new MethodAccessPolicy("Device", 2);
    public static readonly MethodAccessPolicy Both       = new MethodAccessPolicy("Both", 3);
}